tests/components/flipr/test_config_flow.py

```diff
@@ -0,0 +1,166 @@
+"""Test the Flipr config flow."""
+from unittest.mock import patch
+
+import pytest
+from requests.exceptions import HTTPError, Timeout
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.flipr.const import CONF_FLIPR_ID, DOMAIN
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+
+
+@pytest.fixture(name="mock_setup")
+def mock_setups():
+    """Prevent setup."""
+    with patch(
+        "homeassistant.components.flipr.async_setup_entry",
+        return_value=True,
+    ):
+        yield
+
+
+async def test_show_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == config_entries.SOURCE_USER
+
+
+async def test_invalid_credential(hass, mock_setup):
+    """Test invalid credential."""
+    with patch(
+        "flipr_api.FliprAPIRestClient.search_flipr_ids", side_effect=HTTPError()
```

Comment:
Does this patch in the right place?

https://docs.python.org/3/library/unittest.mock.html#where-to-patch